reconcile/aws_ami_share.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from typing import Any, Iterable, Mapping
+
+from reconcile import queries
+
+from reconcile.utils.aws_api import AWSApi
+
+QONTRACT_INTEGRATION = "aws-ami-share"
+MANAGED_TAG = {"Key": "managed_by_integration", "Value": QONTRACT_INTEGRATION}
+
+
+def filter_accounts(accounts: Iterable[dict[str, Any]]) -> list[dict[str, Any]]:
+    sharing_account_names = set()
+    for a in accounts:
+        sharing = a.get("sharing")
+        if sharing:
+            sharing_account_names.add(a["name"])
+            for s in sharing:
+                sharing_account_names.add(s["account"]["name"])
+
+    return [a for a in accounts if a["name"] in sharing_account_names]
+
+
+def get_region(
+    share: Mapping[str, Any],
+    src_account: Mapping[str, Any],
+    dst_account: Mapping[str, Any],
+) -> str:
+    region = share.get("region") or src_account["resourcesDefaultRegion"]
+    if region not in dst_account["supportedDeploymentRegions"]:
+        raise ValueError(f"region {region} is not supported in {dst_account['name']}")
+
+    return region
+
+
+def run(dry_run):
+    accounts = queries.get_aws_accounts(sharing=True)
+    sharing_accounts = filter_accounts(accounts)
+    settings = queries.get_app_interface_settings()
+    aws_api = AWSApi(1, sharing_accounts, settings=settings)
+
+    for src_account in sharing_accounts:
+        sharing = src_account.get("sharing")
```

Comment:
nit: sth I learned from you some weeks ago

```
sharing = src_account.get("sharing") or []
```

Reply:
indeed. but i have learned in the past that it's more efficient to check `if sharing` and not to instantiate an empty list.

it's also more suitable here IMO because it allows us to add more code after the following loop without "missing" that there was no `sharing` section.